CONTRIBUTING.md

```diff
@@ -0,0 +1,388 @@
+
+The following groups are currently supported:
+
+-   **Conditional** \
+    Conditional groups are used to hide/show inputs based on the value of another input. This is useful for inputs that are only relevant in certain situations. E.g. the Save Image node uses conditional groups to hide JPEG-specific options when PNG is selected as the output format.
+
+    We most commonly use `EnumInput`s or `DropDownInput`s as the input that controls the visibility of the group. This is because they are the easiest to work with.
+
+    Example:
+
+    ```python
+    from enum import Enum
+    from nodes.groups import if_enum_group
+    from nodes.properties.inputs import EnumInput, NumberInput
+
+    class Format(Enum):
+        A = 1
+        B = 2
+        C = 3
+
+    # inside @node_group.register(...)
+    inputs = [
+        EnumInput(Format, "Format", default=Format.A).with_id(0),
+        if_enum_group(0, Format.B)(
+            # this weight will only be shown if B is selected
+            NumberInput("B weight", default=50, minimum=0, maximum=100),
+        ),
+    ]
+    ```
+
+-   **Seed** \
+    This group is a wrapper around the `SeedInput` to change its visual appearance for better useability.
+
+    ```python
+    from nodes.group import group
+    from nodes.properties.inputs import SeedInput
+
+    # inside @node_group.register(...)
+    inputs = [
+        seed_group(SeedInput()),
+    ]
+    ```
+
+-   And more. See `backend/src/nodes/groups.py` for more groups.
+
+#### Types
+
+All inputs and outputs have types. These types are automatically generated from the input/output class and the supplied arguments. E.g. `NumberInput("foo", minimum=0, maximum=100)` will have the type `int(0..100)` (read: an integer between 0 and 100 inclusive).
+
+Types are expressed using a custom type system called Navi. Navi is not relative to Python's type hints or TypeScript types, it's a completely separate system. Since the Navi types of inputs and outputs are automatically generated, you rarely have to interact with Navi directly.
+
+### Packages
+
+A package is a collection of nodes. Chainner comes with a set of standard nodes (`chaiNNer_standard`), but you can also create your own packages.
+
+Unfortunately, packages are still a bit of a work in progress. We plan to improve the package system in the future. Right now, they are just fancy directories that contain nodes.
+
+### Categories
+
+Nodes are categorized. Categories are defined per package and follow a strict hierarchy. You can see this hierarchy in the node selector sidebar of chainner.
+
+At the top of the hierarchy are categories (e.g. Image Filter). Each category can have any number of subcategories, called node groups. Node groups then contain the actual nodes.
+
+This gives us a clear path: package -> category -> node group -> node. E.g. the path of the Gaussian Blur node is `chaiNNer_standard -> Image Filter -> Blur -> Gaussian Blur`.
+
+## The anatomy of a node
```

Comment:
Don't you already explain this stuff further up?

Reply:
Because I wanted this section to be independent of Adding a new node. You and I know what a node looks like, but newcomers might not.